Utilities/DataFlow/test/test_PayloadMerger01.cxx

```diff
@@ -0,0 +1,77 @@
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+#include <iostream>
+#include <memory>
+#include "DataFlow/PayloadMerger.h"
+#include "DataFlow/FakeTimeframeBuilder.h"
+#include "DataFlow/TimeframeParser.h"
+#include "DataFlow/SubframeUtils.h"
+#include "fairmq/FairMQTransportFactory.h"
+#include "fairmq/FairMQParts.h"
+#include "fairmq/FairMQDevice.h"
+
+using SubframeId = o2::dataflow::SubframeId;
+using HeartbeatHeader = o2::Header::HeartbeatHeader;
+using HeartbeatTrailer = o2::Header::HeartbeatTrailer;
+
+SubframeId fakeAddition(o2::dataflow::PayloadMerger<SubframeId> &merger,
+                  std::unique_ptr<FairMQTransportFactory> &transport,
+                  int64_t orbit) {
+  // Create a message
+  //
+  // We set orbit to be always the same and the actual contents to be 127
+  static size_t dummyMessageSize = 1000;
+  auto msg = transport->CreateMessage(dummyMessageSize);
+  char *b = reinterpret_cast<char*>(msg->GetData()) + sizeof(HeartbeatHeader);
+  for (size_t i = 0; i < (dummyMessageSize - sizeof(HeartbeatHeader)); ++i) {
+    b[i] = orbit;
+  }
+  b[0] = 127;
+  HeartbeatHeader *header = reinterpret_cast<HeartbeatHeader*>(msg->GetData());
+  header->orbit = orbit;
+  return merger.aggregate(msg);
+}
+
+int main(int argc, char **argv) {
+  std::unique_ptr<FairMQTransportFactory> zmq(FairMQDevice::MakeTransport("zeromq"));
+
+  // Needs two subtimeframe to merge them
+  auto checkIfComplete = [](SubframeId id, o2::dataflow::PayloadMerger<SubframeId>::MessageMap &m) -> bool {
+    if (m.count(id) < 3)
+      return false;
+    return true;
+  };
+
+  // Id is given by the orbit
+  auto makeId = [](std::unique_ptr<FairMQMessage> &msg) {
+    auto header = reinterpret_cast<o2::Header::HeartbeatHeader const*>(msg->GetData());
+    return o2::dataflow::makeIdFromHeader(*header, 0, 2);
+  };
+
+  o2::dataflow::PayloadMerger<SubframeId> merger(makeId, checkIfComplete, o2::dataflow::extractDetectorPayloadStrip);
+  char *finalBuf = new char[3000];
+  size_t finalSize = 0;
+  auto id = fakeAddition(merger, zmq, 1);
+  finalSize = merger.finalise(&finalBuf, id);
+  assert(finalSize == 0); // Not enough parts, not merging yet.
+  id = fakeAddition(merger, zmq, 1);
+  finalSize = merger.finalise(&finalBuf, id);
+  assert(finalSize == 0); // Not enough parts, not merging yet.
+  id = fakeAddition(merger, zmq, 2);
+  finalSize = merger.finalise(&finalBuf, id);
+  assert(finalSize == 0); // Different ID, not merging yet.
+  id = fakeAddition(merger, zmq, 1);
+  finalSize = merger.finalise(&finalBuf, id);
+  assert(finalSize); // Now we merge!
+  size_t partSize = (1000-sizeof(HeartbeatHeader) - sizeof(HeartbeatTrailer));
+  assert(finalSize == 3*partSize); // This should be the calculated size
+  for (size_t i = 0; i < finalSize; ++i)
```

Comment:
according to conventions, there should be a brace `{` even round single line statements